plugins/catalog/src/apis/EntityPresentationApi/DefaultEntityPresentationApi.ts

```diff
@@ -260,47 +260,70 @@ export class DefaultEntityPresentationApi implements EntityPresentationApi {
+        promise: Promise.resolve(initialSnapshot),
+      };
+    }
+
+    const loadingPromise = Promise.resolve().then(() =>
+      this.#loader?.load(entityRef),
+    );
+
     // And then the following snapshot
-    const observable = !needsLoad
-      ? undefined
-      : new ObservableImpl<EntityRefPresentationSnapshot>(subscriber => {
-          let aborted = false;
-
-          Promise.resolve()
-            .then(() => this.#loader?.load(entityRef))
-            .then(newEntity => {
-              if (
-                !aborted &&
-                newEntity &&
-                newEntity.metadata.etag !== entity?.metadata.etag
-              ) {
-                const updatedSnapshot = render({
-                  loading: false,
-                  entity: newEntity,
-                });
-                subscriber.next(updatedSnapshot);
-              }
-            })
-            .catch(() => {
-              // Intentionally ignored - we do not propagate errors to the
-              // observable here. The presentation API should be error free and
-              // always return SOMETHING that makes sense to render, and we have
-              // already ensured above that the initial snapshot was that.
-            })
-            .finally(() => {
-              if (!aborted) {
-                subscriber.complete();
-              }
-            });
-
-          return () => {
-            aborted = true;
-          };
-        });
+    const observable = new ObservableImpl<EntityRefPresentationSnapshot>(
+      subscriber => {
+        let aborted = false;
+
+        loadingPromise
+          .then(newEntity => {
+            if (
+              !aborted &&
+              newEntity &&
+              newEntity.metadata.etag !== entity?.metadata.etag
+            ) {
+              const updatedSnapshot = render({
+                loading: false,
+                entity: newEntity,
+              });
+              subscriber.next(updatedSnapshot);
+            }
+          })
+          .catch(() => {
+            // Intentionally ignored - we do not propagate errors to the
+            // observable here. The presentation API should be error free and
+            // always return SOMETHING that makes sense to render, and we have
+            // already ensured above that the initial snapshot was that.
+          })
+          .finally(() => {
+            if (!aborted) {
+              subscriber.complete();
+            }
+          });
+
+        return () => {
+          aborted = true;
+        };
+      },
+    );
+
+    const promise = loadingPromise
+      .then(newEntity => {
+        if (newEntity && newEntity.metadata.etag !== entity?.metadata.etag) {
+          return render({
+            loading: false,
+            entity: newEntity,
+          });
+        }
+        return initialSnapshot;
+      })
+      .catch(() => initialSnapshot);
 
     return {
       snapshot: initialSnapshot,
       update$: observable,
```

Comment:
question not related to this PR: if this observable emits only a single value what's the real point of having an observable? 🤔

Reply:
It does allow for reactive updates in the future too, when the catalog emits events for changed entities :)

But realistically, it's more about how it's sort of a pattern to have synchronous `snapshot` and then optionally an observable. It's what gives you the most power and you can easily consume that pair in a `useObservable`. The observable also gives the ability to NOT emit any change events at all if the code finds that "eh, the snapshot turned out to be perfectly fine". A promise always "emits".

But yeah. In practice this turns out to not be an often needed power.